sql/2024/04/inaccurate-sizes-attribute-impact.sql

```diff
@@ -0,0 +1,90 @@
+# HTTP Archive query to measure impact of inaccurate sizes attributes per <img> for WordPress sites.
+#
+# WPP Research, Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+CREATE TEMPORARY FUNCTION
+  getImgSizesAccuracy(custom_metrics STRING)
+  RETURNS ARRAY<STRUCT<sizesAbsoluteError INT64,
+  sizesRelativeError FLOAT64,
+  idealSizesSelectedResourceEstimatedPixels INT64,
+  actualSizesEstimatedWastedLoadedPixels INT64,
+  idealSizesSelectedResourceEstimatedBytes FLOAT64,
+  actualSizesEstimatedWastedLoadedBytes FLOAT64>>
+  LANGUAGE js AS '''
+try {
+  const $ = JSON.parse(custom_metrics);
+  let responsiveImages = JSON.parse($.responsive_images);
+  responsiveImages = responsiveImages['responsive-images'];
```

Comment:
I missed this since the invalid JS didn't get flagged by the BigQuery query validation routine. All of this has now been replaced with a pure SQL approach in 7f0b113, per your suggestion.